fix(oxauth): StatService doesn't always correctly flip month

## Describe the issue

StatService doesn't always correctly flip month

```
2025-06-01 00:09:59,078 TRACE [ForkJoinPool.commonPool-worker-5] [org.gluu.oxauth.service.stat.StatService] (StatService.java:92) - updateStat ...  (initialized: true)
2025-06-01 00:09:59,078 TRACE [ForkJoinPool.commonPool-worker-5] [org.gluu.oxauth.service.stat.StatService] (StatService.java:98) - Started updateStat ...
2025-06-01 00:09:59,079 TRACE [ForkJoinPool.commonPool-worker-5] [org.gluu.oxauth.service.stat.StatService] (StatService.java:126) - Stat entry dn: jansId=02-73-A2-00-71-04_202505,ou=202506,ou=stat,o=gluu
2025-06-01 00:09:59,080 TRACE [ForkJoinPool.commonPool-worker-5] [org.gluu.oxauth.service.stat.StatService] (StatService.java:144) - Stat entry is not found in persistence. dn: jansId=02-73-A2-00-71-04_202505,ou=202506,ou=stat,o=gluu
org.gluu.persist.exception.EntryPersistenceException: Failed to find entry: jansId=02-73-A2-00-71-04_202505,ou=202506,ou=stat,o=gluu
        at org.gluu.persist.ldap.impl.LdapEntryManager.find(LdapEntryManager.java:405) ~[gluu-orm-ldap-4.4.2.Final.jar:?]
        at org.gluu.persist.impl.BaseEntryManager.find(BaseEntryManager.java:714) ~[gluu-orm-core-4.4.2.Final.jar:?]
        at org.gluu.persist.impl.BaseEntryManager.find(BaseEntryManager.java:593) ~[gluu-orm-core-4.4.2.Final.jar:?]
        at org.gluu.persist.impl.BaseEntryManager.find(BaseEntryManager.java:580) ~[gluu-orm-core-4.4.2.Final.jar:?]
        at jdk.internal.reflect.GeneratedMethodAccessor65.invoke(Unknown Source) ~[?:?]
        at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
        at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
        at org.jboss.weld.bean.proxy.AbstractBeanInstance.invoke(AbstractBeanInstance.java:38) ~[weld-core-impl-3.1.9.Final.jar:3.1.9.Final]
        at org.jboss.weld.bean.proxy.ProxyMethodHandler.invoke(ProxyMethodHandler.java:106) ~[weld-core-impl-3.1.9.Final.jar:3.1.9.Final]
        at org.gluu.persist.PersistenceEntryManager$EntityManager$1948487569$Proxy$_$$_WeldClientProxy.find(Unknown Source) ~[gluu-orm-core-4.4.2.Final.jar:?]
        at org.gluu.oxauth.service.stat.StatService.setupCurrentEntry(StatService.java:133) ~[classes/:?]
        at org.gluu.oxauth.service.stat.StatService.updateStat(StatService.java:103) ~[classes/:?]
        at org.gluu.oxauth.service.stat.StatService$Proxy$_$$_WeldClientProxy.updateStat(Unknown Source) ~[classes/:?]
        at org.gluu.oxauth.service.stat.StatTimer.process(StatTimer.java:71) ~[classes/:?]
        at org.gluu.oxauth.service.stat.StatTimer$Proxy$_$$_WeldSubclass.process$$super(Unknown Source) ~[classes/:?]
        at jdk.internal.reflect.GeneratedMethodAccessor439.invoke(Unknown Source) ~[?:?]
        at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
        at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
        at org.jboss.weld.interceptor.proxy.TerminalAroundInvokeInvocationContext.proceedInternal(TerminalAroundInvokeInvocationContext.java:51) ~[weld-core-impl-3.1.9.Final.jar:3.1.9.Final]
        at org.jboss.weld.interceptor.proxy.AroundInvokeInvocationContext.proceed(AroundInvokeInvocationContext.java:78) ~[weld-core-impl-3.1.9.Final.jar:3.1.9.Final]
        at org.gluu.service.cdi.async.AsynchronousInterceptor$1.get(AsynchronousInterceptor.java:36) ~[oxcore-service-4.4.2.sp1.jar:?]
        at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1700) ~[?:?]
        at java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1692) ~[?:?]
        at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290) ~[?:?]
        at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020) ~[?:?]
        at java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656) ~[?:?]
        at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594) ~[?:?]
        at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183) ~[?:?]
Caused by: org.gluu.persist.exception.operation.SearchException: Failed to lookup entry by DN: 'jansId=02-73-A2-00-71-04_202505,ou=202506,ou=stat,o=gluu'
        at org.gluu.persist.ldap.operation.impl.LdapOperationServiceImpl.lookupImpl(LdapOperationServiceImpl.java:629) ~[gluu-orm-ldap-4.4.2.Final.jar:?]
        at org.gluu.persist.ldap.operation.impl.LdapOperationServiceImpl.lookup(LdapOperationServiceImpl.java:605) ~[gluu-orm-ldap-4.4.2.Final.jar:?]
        at org.gluu.persist.ldap.impl.LdapEntryManager.find(LdapEntryManager.java:399) ~[gluu-orm-ldap-4.4.2.Final.jar:?]
        ... 27 more
2025-06-01 00:09:59,082 TRACE [ForkJoinPool.commonPool-worker-5] [org.gluu.oxauth.service.stat.StatService] (StatService.java:147) - Current entry before nullity check, dn jansId=02-73-A2-00-71-04_202505,ou=202505,ou=stat,o=gluu
2025-06-01 00:09:59,082 TRACE [ForkJoinPool.commonPool-worker-5] [org.gluu.oxauth.service.stat.StatService] (StatService.java:163) - Current entry dn jansId=02-73-A2-00-71-04_202505,ou=202505,ou=stat,o=gluu
2025-06-01 00:09:59,083 TRACE [ForkJoinPool.commonPool-worker-5] [org.gluu.oxauth.service.stat.StatService] (StatService.java:113) - Updating entry dn jansId=02-73-A2-00-71-04_202505,ou=202505,ou=stat,o=gluu
2025-06-01 00:09:59,087 TRACE [ForkJoinPool.commonPool-worker-5] [org.gluu.oxauth.service.stat.StatService] (StatService.java:116) - Finished updateStat.

```

## Steps To Reproduce

Investigate stat records on month flip.


## Expected behavior
It should switch cleanly.

## Actual behavior
There are exceptions

Support: `#64`

--- a/oxAuth/Server/src/main/java/org/gluu/oxauth/service/stat/StatService.java
+++ b/oxAuth/Server/src/main/java/org/gluu/oxauth/service/stat/StatService.java
@@ -49,7 +49,6 @@ public class StatService {
     private StaticConfiguration staticConfiguration;
 
     private String nodeId;
-    private String monthlyDn;
     private StatEntry currentEntry;
     private HLL hll;
     private ConcurrentMap<String, Map<String, Long>> tokenCounters;
@@ -74,13 +73,10 @@ public class StatService {
                 return false;
             }
 
-            final Date now = new Date();
-            prepareMonthlyBranch(now);
-            log.trace("Monthly branch created: " + monthlyDn);
-
-            setupCurrentEntry(now);
-            log.info("Initialized Stat Service");
+            prepareMonthlyBranch();
+            setupCurrentEntry();
             initialized = true;
+            log.info("Initialized Stat Service");
             return true;
         } catch (Exception e) {
             log.error("Failed to initialize Stat Service.", e);
@@ -89,53 +85,71 @@ public class StatService {
     }
 
     public void updateStat() {
+        log.trace("updateStat ...  (initialized: {})", initialized);
+
         if (!initialized) {
             return;
         }
 
         log.trace("Started updateStat ...");
 
-        Date now = new Date();
-        prepareMonthlyBranch(now);
-
-        setupCurrentEntry(now);
+        prepareMonthlyBranch();
+        setupCurrentEntry();
 
         final Stat stat = currentEntry.getStat();
         stat.setTokenCountPerGrantType(tokenCounters);
-        stat.setLastUpdatedAt(now.getTime());
+        stat.setLastUpdatedAt(System.currentTimeMillis());
 
         synchronized (hll) {
             currentEntry.setUserHllData(Base64.getEncoder().encodeToString(hll.toBytes()));
         }
+
+        log.trace("Updating entry dn {}", currentEntry.getDn());
         entryManager.merge(currentEntry);
 
         log.trace("Finished updateStat.");
     }
 
-    private void setupCurrentEntry() {
-        setupCurrentEntry(new Date());
+    public static String currentMonth() {
+        return PERIOD_DATE_FORMAT.format(new Date());
     }
 
-    private void setupCurrentEntry(Date now) {
-        final String month = PERIOD_DATE_FORMAT.format(now);
-        String dn = String.format("jansId=%s,%s", nodeId, monthlyDn); // jansId=<id>,ou=yyyyMM,ou=stat,o=gluu
+    public String currentMonthDn() {
+        final String baseDn = getBaseDn();
+        final String month = currentMonth();
+        return String.format("ou=%s,%s", month, baseDn);
+    }
 
-        if (currentEntry != null && month.equals(currentEntry.getStat().getMonth())) {
+    private void setupCurrentEntry() {
+        String currentMonth = currentMonth();
+        String dn = String.format("jansId=%s,%s", nodeId, currentMonthDn()); // jansId=<id>,ou=yyyyMM,ou=stat,o=gluu
+        log.trace("Stat entry dn: {}", dn);
+
+        final boolean sameMonth = currentEntry != null && currentMonth.equals(currentEntry.getStat().getMonth());
+        if (sameMonth) {
+            log.trace("Same month {}", currentMonth);
             return;
+        } else {
+            log.trace("Different month {}", currentMonth);
+            currentEntry = null; // set current entry to null to force re-fetch it from DB or create new one
         }
 
         try {
             StatEntry entryFromPersistence = entryManager.find(StatEntry.class, dn);
-            if (entryFromPersistence != null && month.equals(entryFromPersistence.getStat().getMonth())) {
+            if (entryFromPersistence != null && currentMonth.equals(entryFromPersistence.getStat().getMonth())) {
                 hll = HLL.fromBytes(Base64.getDecoder().decode(entryFromPersistence.getUserHllData()));
                 tokenCounters = new ConcurrentHashMap<>(entryFromPersistence.getStat().getTokenCountPerGrantType());
                 currentEntry = entryFromPersistence;
-                log.trace("Stat entry loaded.");
+                log.trace("Stat entry {} loaded.", dn);
                 return;
+            } else {
+                log.trace("Month does not match. Current month {}, entry month {}, entry dn: {}", currentMonth, entryFromPersistence != null ? entryFromPersistence.getStat().getMonth() : "", dn);
             }
         } catch (EntryPersistenceException e) {
-            log.trace("Stat entry is not found in persistence.");
+            log.trace("Stat entry is not found in persistence. dn: " + dn, e);
         }
+
+        log.trace("Current entry before nullity check, dn {}", currentEntry != null ? currentEntry.getDn() : "null");
 
         if (currentEntry == null) {
             log.trace("Creating stat entry ...");
@@ -150,6 +164,8 @@ public class StatService {
             entryManager.persist(currentEntry);
             log.trace("Created stat entry. nodeId:" + nodeId);
         }
+
+        log.trace("Current entry dn {}", currentEntry != null ? currentEntry.getDn() : "null");
     }
 
     public HLL newHll() {
@@ -158,6 +174,7 @@ public class StatService {
 
     private void initNodeId() {
         if (StringUtils.isNotBlank(nodeId)) {
+            log.trace("NodeId is not blank: {}", nodeId);
             return;
         }
 
@@ -173,6 +190,7 @@ public class StatService {
         } catch (Exception e) {
             log.error("Failed to identify nodeId.", e);
             nodeId = UUID.randomUUID().toString();
+            log.trace("NodeId created: " + nodeId);
         }
     }
 
@@ -184,22 +202,23 @@ public class StatService {
         return staticConfiguration.getBaseDn().getStat();
     }
 
-    private void prepareMonthlyBranch(Date now) {
-        final String baseDn = getBaseDn();
-        final String month = PERIOD_DATE_FORMAT.format(now); // yyyyMM
-        monthlyDn = String.format("ou=%s,%s", month, baseDn); // ou=yyyyMM,ou=stat,o=gluu
-
-        if (!entryManager.hasBranchesSupport(baseDn)) {
+    private void prepareMonthlyBranch() {
+        if (!entryManager.hasBranchesSupport(getBaseDn())) {
+            log.trace("Monthly branch creation is skipped. DB does not support branches.");
             return;
         }
 
+        String monthlyDn = currentMonthDn();
+
         try {
             if (!entryManager.contains(monthlyDn, SimpleBranch.class)) { // Create ou=yyyyMM branch if needed
-                createBranch(monthlyDn, month);
+                createBranch(monthlyDn, currentMonth());
+                log.info("Monthly branch is created: {}", monthlyDn);
             }
         } catch (Exception e) {
-            log.error("Failed to prepare monthly branch: " + monthlyDn, e);
-            throw e;
+            final String msg = "Failed to prepare monthly branch: " + monthlyDn;
+            log.error(msg, e);
+            throw new RuntimeException(msg, e);
         }
     }
 

